registry/distribution/manifest.go

```diff
@@ -0,0 +1,46 @@
+/*
+Copyright The ORAS Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package distribution
+
+import (
+	"strings"
+
+	ocispec "github.com/opencontainers/image-spec/specs-go/v1"
+	artifactspec "github.com/oras-project/artifacts-spec/specs-go/v1"
+	"oras.land/oras-go/v2/internal/docker"
+)
+
+// manifestMediaTypes contains the media type of manifests.
+var manifestMediaTypes = []string{
+	docker.MediaTypeManifest,
+	docker.MediaTypeManifestList,
+	ocispec.MediaTypeImageManifest,
+	ocispec.MediaTypeImageIndex,
+	artifactspec.MediaTypeArtifactManifest,
```

Comment:
Adding new media types will be as simple as adding the new media types into the `manifestMediaTypes` array. The registry client does not do any deserialization on the manifests so that any manifest is accepted, even it is a binary.

However, to make the new manifest recogizable, the `DownEdges()` function in the `content/graph.go` file needs to be updated.

Reply:
With 747c1c3, adding new manifest media types is as simple as configure `Repository.ManifestMediaTypes`.

To make the new manifest recogizable, the `DownEdges()` function in the `content/graph.go` file still needs to be updated. Or, we can have more options to replace `DownEdges()` in `Copy()` by #59 .